{{cookiecutter.github_repository}}/{{cookiecutter.main_module}}/users/auth/serializers.py

```diff
@@ -43,3 +39,58 @@ def get_name(self, obj):
 
     def get_auth_token(self, obj):
         return get_token_for_user(obj, "authentication")
+
+
+class PasswordChangeSerializer(serializers.Serializer):
+    current_password = serializers.CharField(required=True)
+    new_password = serializers.CharField(required=True)
+
+    default_error_messages = {
+        'invalid_password': 'Current password does not match'
+    }
+
+    def validate_current_password(self, value):
+        if not self.context['request'].user.check_password(value):
+            raise serializers.ValidationError(self.default_error_messages['invalid_password'])
+        return value
+
+    def validate_new_password(self, value):
+        # https://docs.djangoproject.com/en/2.0/topics/auth/passwords/#django.contrib.auth.password_validation.validate_password
+        password_validation.validate_password(value)
+        return value
+
+
+class PasswordResetSerializer(serializers.Serializer):
+    email = serializers.EmailField(required=True)
+
+
+class PasswordResetConfirmSerializer(serializers.Serializer):
+    new_password = serializers.CharField(required=True)
+    uid = serializers.CharField(required=True)
+    token = serializers.CharField(required=True)
+
+    default_error_messages = {
+        'invalid_uid': 'Invalid or malformed uid',
+        'invalid_token': 'Invalid token or the token has expired'
+    }
+
+    def validate_new_password(self, value):
+        password_validation.validate_password(value)
+        return value
+
+    def validate_uid(self, value):
+        """
+        Validate the encoded user id and add a user attribute to the serializer.
+        The user attribute is used in the validate_token method and the api view.
+        """
+        try:
+            user_id = decode_uid(value)
+            self.user = get_user_model().objects.get(id=user_id)
+        except (get_user_model().DoesNotExist, ValueError, TypeError, OverflowError):
```

Comment:
https://github.com/encode/django-rest-framework/issues/5197

The app will raise an internal server error if an invalid uuid is decoded from uid.
@theskumar @jainmickey